mineral-ui/mineral-ui
# [Dialog] New component

### Description

- Add focus-trap-react dependency
- Add internal close icon
- Refactor Table to extract shared OverflowContainer and add spec
- Refactor Table to extract shared textWithThemeOverrides and add spec
- Add children and capitalize utils
- Refactor Dropdown, Popover, and Portal with various improvements
- Add OverflowContainerWithShadows and add spec
- Add Dialog components
- Refactor Dialog, Popover, and Select to provide helpful Popper modifiers via Context
- Update website with routes, demos, and examples for Dialog
- Add Dialog specs and snapshots

### Motivation and context

Closes #527 

### Screenshots, videos, or demo, if appropriate


https://527-dialog--mineral-ui.netlify.com/components/dialog

### How to test


1. `cd mineral-ui && npm start`
2. Navigate to [Dialog](https://527-dialog--mineral-ui.netlify.com/components/dialog) and its subpages
3. Confirm everything works and that the content is useful, understandable, and error-free

### Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change
- Other (provide details below)

Breaking: Table theme variable renamed
Other: Refactors (see description)

### Checklist




### How does this PR make you feel?

![Willy Wonka tell me more](https://media.giphy.com/media/xT9IgFlWsUMDFzVIu4/giphy.gif)
